src/modules/languages/rust.nix

```diff
@@ -49,13 +46,11 @@ in
 
   config = lib.mkMerge [
     (lib.mkIf cfg.enable {
-      packages = attrValues (getAttrs tools cfg.packages) ++ lib.optional pkgs.stdenv.isDarwin pkgs.libiconv;
+      packages = [ cfg.package ] ++ lib.optional pkgs.stdenv.isDarwin pkgs.libiconv;
 
       # enable compiler tooling by default to expose things like cc
       languages.c.enable = lib.mkDefault true;
 
-      env.RUST_SRC_PATH = cfg.packages.rust-src;
-
```

Comment:
i didn't know what the best way to set this was, but I think this is fine anyways? `cargo build` still works and i verified `rustc --print sysroot` returns a satisfactory directory

Reply:
`RUST_SRC_PATH` is necessary when `rust-src` is not at the same location as as `rustc`. This is the case with the rust toolchain from `nixpkgs`. You'll run into this issue when using rust-analyzer in vscode.

Reply:
i thought this would work since i'm using `symlinkJoin`? the symlinks are in the same location, does `rustc` dereference them?

Reply:
Unfortunately not. It's based on the physical location of `librustc_driver`. See: https://github.com/rust-lang/rust/pull/103660#issuecomment-1307721558

Reply:
ah, quite unfortunate :/